### PR TITLE
feat(SHP-2124): Create Fulfilled Order

### PR DESCRIPTION
**Pull request by RohanLeach**

### CODE REVIEW

File: orders.go [hunk 1]
```diff
@@ -75,6 +75,8 @@ type Order struct {
 	NoteAttributes NoteAttributes
 	// OrderNumber tells you that this is the shop's nth order.
 	OrderNumber int
+	// orderStatusURL to find the order status
```
Comment (JamesFowke) [inline orders.go:78]:
```suggestion
	// OrderStatusURL to find the order status
```

File: orders.go [hunk 2]
```diff
@@ -75,6 +75,8 @@ type Order struct {
 	NoteAttributes NoteAttributes
 	// OrderNumber tells you that this is the shop's nth order.
 	OrderNumber int
+	// OrderStatusURL to find the order status
```
Comment (nichharp-mohc) [inline orders.go:78]:
```suggestion
	// OrderStatusURL is the Thank You page (Shopify doubles it as a order status page)
```
Note that there's 2 URLs for this sorta thing. There's a legit thank you page with `thank_you` in the URL and then this one.
The user is only shown the thank you page once, when it's attempted to be hit again the user is redirected to the OrderStatusURL.
This is why tracking on this page might get complicated. We need that spike I mentioned before on how we can track this.

File: orders.go [hunk 3]
```diff
@@ -191,6 +193,8 @@ type OrderRepository interface {
 	Open(id int64) (Order, error)
 	// Create creates a new order
 	Create(order Order) (Order, error)
+	// Create creates a new order
+	CreateSkipCheckout(order Order) (Order, error)
```
Comment (nichharp-mohc) [inline orders.go:197]:
```suggestion
	// Create creates a new fulfilled order
	CreateFulfilled(order Order) (Order, error)
```

File: orders.go [hunk 4]
```diff
@@ -75,6 +75,8 @@ type Order struct {
 	NoteAttributes NoteAttributes
 	// OrderNumber tells you that this is the shop's nth order.
 	OrderNumber int
+	// OrderStatusURL is the Thank You page (Shopify doubles it as a order status page)
```
Comment (nichharp-mohc) [inline orders.go:78]:
```suggestion
	// OrderStatusURL the URL pointing to the order status web page, if applicable.
```

**Review by nichharp-mohc — CHANGES_REQUESTED** [review]

**Review by ezelopes-mohc — APPROVED** [review]

**Review by nichharp-mohc — APPROVED** [review]

**Pull request merged**